db/blip_handler.go

```diff
@@ -124,6 +127,32 @@ func (bh *blipHandler) refreshUser() error {
 	return nil
 }
 
+// userBlipHandler wraps another blip handler to specify a collection
+func collectionBlipHandler(next blipHandlerFunc) blipHandlerFunc {
+	return func(bh *blipHandler, bm *blip.Message) error {
+		collectionIndexStr, ok := bm.Properties[BlipCollection]
+		if !ok {
+			bh.collection = bh.db
+			return next(bh, bm)
+		}
+		collectionIndex, err := strconv.Atoi(collectionIndexStr)
+		if err != nil {
+			return base.HTTPErrorf(http.StatusBadRequest, "collection property needs to be an int, was %q", collectionIndexStr)
+		}
+
+		if len(bh.collectionMapping) == 0 {
+			return base.HTTPErrorf(http.StatusBadRequest, "Passing collection requires calling GetCollections first")
+		}
+
+		if len(bh.collectionMapping) <= collectionIndex {
+			return base.HTTPErrorf(http.StatusBadRequest, "Collection index %d does not match a collection set by GetCollections", collectionIndex)
+		}
+		bh.collection = bh.collectionMapping[collectionIndex]
```

Comment:
as discussed, check if the mapped collection is nil here and return an appropriate error (404?) to avoid panicing inside a handler trying to use an invalid collection